### PR TITLE
More misc fuzz fixes

### PR DESCRIPTION
**Pull request by kripken**
Almost the end of the fuzzing work. This is a bunch of tiny fixes fuzzing found. At this point, both binary and text parsers are hardened, and the fuzzer runs for days without finding new stuff.

### CODE REVIEW

File: src/wasm/wasm-s-parser.cpp [hunk 1]
```diff
@@ -1614,7 +1619,7 @@ void SExpressionWasmBuilder::parseImport(Element& s) {
       }
       if (inner.size() > j+1) {
         Element& result = *inner[j+1];
-        assert(result[0]->str() == RESULT);
+        if (result[0]->str() != RESULT)  throw ParseException("expected result");
```
Comment (jgravelle-google) [inline src/wasm/wasm-s-parser.cpp]: Nit: Double-spacing before `throw`

As an aside: yay for changing asserts to errors with messages!

File: src/wasm/wasm-s-parser.cpp [hunk 2]
```diff
@@ -577,6 +577,7 @@ void SExpressionWasmBuilder::parseFunction(Element& s, bool preParseImport) {
     im->kind = ExternalKind::Function;
     im->functionType = wasm.getFunctionType(type)->name;
     wasm.addImport(im.release());
+    if (currFunction) throw ParseException("import module inside function dec");
```
Comment (jgravelle-google) [inline src/wasm/wasm-s-parser.cpp]: Wait what?
If `currFunction` we `throw`, but if `!currFunction` we `assert`?

File: src/wasm/wasm-s-parser.cpp [hunk 3]
```diff
@@ -1650,20 +1650,36 @@ void SExpressionWasmBuilder::parseImport(Element& s) {
     }
   } else if (im->kind == ExternalKind::Table) {
     if (j < inner.size() - 1) {
-      wasm.table.initial = atoi(inner[j++]->c_str());
+      uint64_t num = atoi(inner[j++]->c_str());
```
Comment (jgravelle-google) [inline src/wasm/wasm-s-parser.cpp]: 4x reuse, how about extracting this to `uint64_t checkedTableSize(char*, Element&);`

And/or, they all say "excessive table size", when two of them are memories.

Comment (jgravelle-google) [inline src/wasm/wasm-s-parser.cpp]: Also should this use `atoll`?

File: src/wasm/wasm-s-parser.cpp [hunk 4]
```diff
@@ -1460,7 +1460,11 @@ void SExpressionWasmBuilder::parseMemory(Element& s, bool preParseImport) {
       return;
     }
   }
-  wasm.memory.initial = atoi(s[i++]->c_str());
+  uint64_t num = atoi(s[i++]->c_str());
```
Comment (jgravelle-google) [inline src/wasm/wasm-s-parser.cpp]: More candidates for reuse + `atoll`

File: src/wasm/wasm-s-parser.cpp (hunk 2 above)
Comment (kripken) [inline src/wasm/wasm-s-parser.cpp]: The assert is redundant, yeah, I'll remove it.

File: src/wasm/wasm-s-parser.cpp (hunk 3 above)
Comment (kripken) [inline src/wasm/wasm-s-parser.cpp]: Good idea, refactoring to `getCheckedAddress` for those, and use `atoll`.

**Pull request merged**